feat(data-orchestrator): add how to create workflow MTA-7013 (#6438)
scaleway/docs-content: pages/data-orchestrator/how-to/create-workflow.mdx

---
title: How to create a workflow using Data Orchestrator
description: This page explains how to create a Data Orchestrator workflow on the Scaleway console.
tags: data-orchestrator workflow
dates:
  posted: 2026-04-01
---
import Requirements from '@macros/iam/requirements.mdx'
import UploadWorkflow from '@macros/data-orchestrator/upload-workflow.mdx'

<Message type="note">
  Data Orchestrator is currently in Private Beta. During this Beta period, the workflows only run Scaleway Serverless Jobs.
</Message>

<Requirements />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/iam/concepts/#owner) status or [IAM permissions](/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- A [Container Registry namespace](/container-registry/how-to/create-namespace/)
- An image [pushed](/container-registry/how-to/push-images/) to your namespace (for task execution)
- Created at least three [Serverless Jobs](/serverless-jobs/how-to/create-job/)
- Basic familiarity with YAML syntax
- Basic familiarity with [Serverless workflow.io](https://serverlessworkflow.io/) language

## Create a workflow definition YAML file

> Data Orchestrator requires the user to upload a YAML file to use as workflow definition.

Reviewer: ```suggestion
Data Orchestrator requires the user to upload a YAML file to use as a workflow definition.
```

<Message type="tip">

>   This is a basic example of a sequential workflow, for more examples, see LINK.

Reviewer: LINK --> Replace placeholder with an actual link.

    do:
    - task1:
        call: "serverless_job"
        metadata:
          description: "The first job"
        with:
          id: "<Serverless_job1_ID>"
          region: "fr-par"
    - task2:
        call: "serverless_job"
        metadata:
          description: "The second job"
        with:
          id: "<Serverless_job2_ID>"
          region: "fr-par"
    - task3:
        call: "serverless_job"
        metadata:
          description: "The third job"
        with:
          id: "<Serverless_job3_ID>"
          region: "fr-par"
    ```
2. Replace the ID placeholders with the IDs of your Serverless Jobs. You can copy them from the [Serverless Jobs page](https://console.scaleway.com/serverless-jobs/).
    <Message type="note">
      The fields `namespace` and `name` are required. They can be used as metadata to help organize your YAML files.
    </Message>
3. Save the YAML file.

## Create a Data Orchestrator workflow

1. Click **Data Orchestrator** in the **Data & Analytics** section of the [console](https://console.scaleway.com/) side menu. The list of your workflows displays.
2. Click **+ Create workflow**. The Data Orchestrator workflow creation wizard displays.

> 3. Select the region in which your workflow will be created.

Reviewer: ```suggestion
3. Select the region in which you want to create your workflow.
```

Author: For consistency, I reused this form that appears a lot in the documentation.
